src/Microsoft.ML.OnnxTransform/OnnxUtils.cs

```diff
@@ -81,11 +81,13 @@ public OnnxModel(string modelFile, int? gpuDeviceId = null, bool fallbackToCpu =
         {
             _modelFile = modelFile;
 
-            if (gpuDeviceId.HasValue)
+            // Reserve Int32.MinValue for test purpose.
```

Comment:
Why can't we use `null` instead?

Reply:
changed to null.

---
In reply to: [251942603](https://github.com/dotnet/machinelearning/pull/2085#discussion_r251942603) [](ancestors = 251942603)

Reply:
I meant, why do we need to have `OnnxTransformer.NullGpuID`.  Why can't we use the literal `null` for the `int?` parameter?

---
In reply to: [252049662](https://github.com/dotnet/machinelearning/pull/2085#discussion_r252049662) [](ancestors = 252049662,251942603)

Reply:
The command line parser test (below), requires values for all input parameters, even for nullable parameters (not sure if this is by design or not). However, for the nullable input parameters, there's doesn't seem to be a way to pass a value which represents 'null'.  Since we can't pass in a null value in the particular test, I needed a non-null value which would not actually try to run on Gpu.  Let me know if there's a way to pass in a 'null' value in the command line parser test -- that would obviate the need for this variable.

(https://github.com/dotnet/machinelearning/blob/master/test/Microsoft.ML.OnnxTransformTest/OnnxTransformTests.cs#L243)

Reply:
It's because the property has `ArgumentType.Required` on its attribute:

https://github.com/dotnet/machinelearning/blob/0c385eee4b2bc532ae90b9d4fe913d18547f96ac/src/Microsoft.ML.OnnxTransform/OnnxTransform.cs#L73-L74

Reply:
Update the attribute to [Argument(ArgumentType.AtMostOnce | *ArgumentType.AtMostOnce*,...]
And removed the NullGpu now.